contracts/libraries/Pool.sol

```diff
@@ -120,25 +128,36 @@ library Pool {
         self.slot0 = Slot0({
             sqrtPriceX96: sqrtPriceX96,
             tick: tick,
-            protocolSwapFee: protocolSwapFee,
-            hookSwapFee: hookSwapFee,
-            protocolWithdrawFee: protocolWithdrawFee,
-            hookWithdrawFee: hookWithdrawFee
+            protocolFees: _calculateFeesFromSwapAndWithdraw(protocolSwapFee, protocolWithdrawFee),
+            hookFees: _calculateFeesFromSwapAndWithdraw(hookSwapFee, hookWithdrawFee)
         });
     }
 
-    function setProtocolFees(State storage self, uint8 newProtocolSwapFee, uint8 newProtocolWithdrawFee) internal {
+    function getSwapFee(uint24 feesStorage) internal pure returns (uint16) {
+        return uint16(feesStorage >> 12);
+    }
+
+    function getWithdrawFee(uint24 feesStorage) internal pure returns (uint16) {
+        return uint16(feesStorage & 0xFFF);
+    }
+
+    function setProtocolFees(State storage self, uint16 newProtocolSwapFee, uint16 newProtocolWithdrawFee) internal {
         if (self.slot0.sqrtPriceX96 == 0) revert PoolNotInitialized();
 
-        self.slot0.protocolSwapFee = newProtocolSwapFee;
-        self.slot0.protocolWithdrawFee = newProtocolWithdrawFee;
+        self.slot0.protocolFees = _calculateFeesFromSwapAndWithdraw(newProtocolSwapFee, newProtocolWithdrawFee);
     }
 
-    function setHookFees(State storage self, uint8 newHookSwapFee, uint8 newHookWithdrawFee) internal {
+    function setHookFees(State storage self, uint16 newHookSwapFee, uint16 newHookWithdrawFee) internal {
         if (self.slot0.sqrtPriceX96 == 0) revert PoolNotInitialized();
 
-        self.slot0.hookSwapFee = newHookSwapFee;
-        self.slot0.hookWithdrawFee = newHookWithdrawFee;
+        self.slot0.hookFees = _calculateFeesFromSwapAndWithdraw(newHookSwapFee, newHookWithdrawFee);
+    }
+
+    function _calculateFeesFromSwapAndWithdraw(uint16 swapFee, uint16 withdrawFee) internal pure returns (uint24) {
+        if (swapFee > FEE_MAX) revert FeeOutOfBounds(swapFee);
```

Comment:
After making those other changes, since the fee will not be passed as a uint16, you don't need to check for FeeOutOfBounds because they cannot exceed the 24 bits allocated. Also FEE_MAX is misleading because the actual denominator that a protocol fee is allowed to be is already defined by MIN_PROTOCOL_FEE_DENOMINATOR

Reply:
Don't think this function is needed at all anymore